overlord/overlord_test.go

```diff
@@ -1399,3 +1400,25 @@ func (ovs *overlordSuite) TestLockWithTimeoutFailed(c *C) {
 		"EXTEND_TIMEOUT_USEC=5000",
 	})
 }
+
+func (ovs *overlordSuite) TestEnsureLoopLogging(c *C) {
+	entries, err := os.ReadDir(".")
+	c.Assert(err, IsNil)
+
+	for _, entry := range entries {
+		if entry.IsDir() && strings.HasSuffix(entry.Name(), "state") {
+			testFilePath := filepath.Join(entry.Name(), entry.Name()+"_test.go")
+
+			if _, err := os.Stat(testFilePath); err == nil {
+				mgrFiles, err := filepath.Glob(filepath.Join(entry.Name(), "*mgr.go"))
+				c.Assert(err, IsNil)
+
+				if len(mgrFiles) > 0 {
+					content, err := os.ReadFile(testFilePath)
+					c.Assert(err, IsNil)
+					c.Assert(strings.Contains(string(content), "TestEnsureLoopLogging"), Equals, true, Commentf("File %s does not contain a TestEnsureLoopLogging test."))
+				}
+			}
+		}
+	}
```

Comment:
This is a bit of a nitpicky suggestion so feel free to ignore. Since the globbing of `*mgr.go` will at most only pick up one file and it's always the dir name minus the "state" suffix, maybe we can just check for the presence of that. Same for the error check on the Stat. Maybe we can express it more explicitly and use `TrimSuffix` and `FileExists` to do some of the heavy lifting:

```suggestion
for _, entry := range entries {
		if !(entry.IsDir() && strings.HasSuffix(entry.Name(), "state") && entry.Name() != "state") {
			continue
		}

		testPath := filepath.Join(entry.Name(), entry.Name()+"_test.go")
		prefix := strings.TrimSuffix(entry.Name(), "state")
		mgrPath := filepath.Join(entry.Name(), prefix+"mgr.go")
		if !osutil.FileExists(testPath) || !osutil.FileExists(mgrPath) {
			continue
		}

		content, err := os.ReadFile(testPath)
		c.Assert(err, IsNil)
		c.Assert(strings.Contains(string(content), "TestEnsureLoopLogging"), Equals, true, Commentf("File %s does not contain a TestEnsureLoopLogging test.", testPath))
	}
```